tests/integration/raw_studies_blueprint/test_fetch_raw_data.py

```diff
@@ -86,19 +87,64 @@ def test_get_study(
                 params={"path": f"/{rel_path.as_posix()}", "depth": 1},
                 headers=headers,
             )
-            res.raise_for_status()
+            assert res.status_code == 200, res.json()
             actual = res.content
             expected = file_path.read_bytes()
             assert actual == expected
 
+        # If you try to retrieve a file that doesn't exist, we should have a 404 error
+        res = client.get(
+            f"/v1/studies/{study_id}/raw",
+            params={"path": "user/somewhere/something.txt"},
+            headers=headers,
+        )
+        assert res.status_code == 404, res.json()
+        assert res.json() == {
+            "description": "'somewhere' not a child of User",
+            "exception": "ChildNotFoundError",
+        }
+
+        # If you want to update an existing resource, you can use PUT method.
+        # But, if the resource doesn't exist, you should have a 404 Not Found error.
+        res = client.put(
+            f"/v1/studies/{study_id}/raw",
+            params={"path": "user/somewhere/something.txt"},
+            headers=headers,
+            files={"file": io.BytesIO(b"Goodbye World!")},
+        )
+        assert res.status_code == 404, res.json()
+        assert res.json() == {
+            "description": "'somewhere' not a child of User",
+            "exception": "ChildNotFoundError",
+        }
+
+        # To create or update a resource, you can use PUT method and the `create_missing` flag.
+        # The expected status code should be 204 No Content.
+        res = client.put(
+            f"/v1/studies/{study_id}/raw",
+            params={"path": "user/somewhere/something.txt", "create_missing": True},
+            headers=headers,
+            files={"file": io.BytesIO(b"Goodbye Cruel World!")},
+        )
+        assert res.status_code == 204, res.json()
+
+        # You can check that the resource has been created or updated.
+        res = client.get(
+            f"/v1/studies/{study_id}/raw",
+            params={"path": "user/somewhere/something.txt"},
+            headers=headers,
+        )
+        assert res.status_code == 200, res.json()
+        assert res.content == b"Goodbye Cruel World!"
+
```

Comment:
You could probably add a test with the flag to `True` and a path leading to an existing file.

Reply:
This is exactly what is it done in the previous test, because the default value of `create_missing` is `False`:

```python
# If you want to update an existing resource, you can use PUT method.
# But, if the resource doesn't exist, you should have a 404 Not Found error.
res = client.put(
    f"/v1/studies/{study_id}/raw",
    params={"path": "user/somewhere/something.txt"},
    headers=headers,
    files={"file": io.BytesIO(b"Goodbye World!")},
)
assert res.status_code == 404, res.json()
assert res.json() == {
    "description": "'somewhere' not a child of User",
    "exception": "ChildNotFoundError",
}
```

Reply:
Done